### PR TITLE
Upgrade Kotlin and AndroidX dependencies

### PR DESCRIPTION
**Pull request by msfjarvis**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [x] Dependency updates

## :scroll: Description

Performs the following dependency upgrades

- AGP: 4.1.3 -> 4.2.0
- Dokka: 1.4.30 -> 1.4.32
- AndroidX Test: 1.4.0-alpha05 -> 1.4.0-alpha06
- ConstraintLayout: 2.1.0-beta01 -> 2.1.0-beta02
- ModernAndroidPreferences: 2.0 -> 2.1.0
- sshauthentication-api: 5.7.1 -> 5.7.5

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -12,6 +12,16 @@ repositories {
   google()
   gradlePluginPortal()
   mavenCentral()
+  maven("https://maven.pkg.jetbrains.space/public/p/kotlinx-html/maven") {
```
Comment (fmeum) [inline buildSrc/build.gradle.kts:17]: Could you say what we are using this for? It looks exciting.

File: build.gradle.kts [hunk 2]
```diff
@@ -5,6 +5,7 @@
 plugins {
   `binary-compatibility-validator`
   `aps-plugin`
+  id("dev.zacsweers.kgp-150-leak-patcher") version "1.0.1"
```
Comment (fmeum) [inline build.gradle.kts:9]: Is the underlying leak being tracked somewhere?

**Review by fmeum — APPROVED** [review]

File: build.gradle.kts (hunk 2 above)
Comment (msfjarvis) [reply build.gradle.kts:9]: Yep, as [KT-46368](https://youtrack.jetbrains.com/issue/KT-46368)

File: buildSrc/build.gradle.kts (hunk 1 above)
Comment (msfjarvis) [reply buildSrc/build.gradle.kts:17]: It's for [Dokka](https://github.com/Kotlin/dokka/releases/tag/v1.4.32), I'll add a comment.

**Review by fmeum — APPROVED** [review]

**Pull request merged**